#ifndef CANOPENTESTDLG_H
#define CANOPENTESTDLG_H

#include <QWidget>
#include <qdialog.h>
#include "ui_canopentestdlg.h"
#include "canopenprotocol.h"

class CanOpenTestDlg : public QDialog
{
    Q_OBJECT

public:
    CanOpenTestDlg(QDialog *parent = 0);
    ~CanOpenTestDlg();

private:
    Ui::CanOpenTestDlg ui;
    CanOpenProtocol *m_canOpen;
    static const int m_cardIndex = 0;
    static const int m_canNode = 0;
    int m_countRevMSG;
    QTimer *timer;
    void init();
protected:
	void closeEvent(QCloseEvent *);
public slots:
	void on_runPushbutton();
	void on_stopPushButton();
	void on_clearMsg();
	void on_transmitData();
	void on_recieveData();
    void on_showCardStatus();
};

#endif // CANOPENTESTDLG_H
